system-test/bigquery.ts

```diff
@@ -1698,14 +1698,21 @@ describe('BigQuery', () => {
       filter: `labels.${GCLOUD_TESTS_PREFIX}`,
     });
 
-    const deleteDatasetPromises = datasets
-      .filter(async dataset => {
-        const [metadata] = await dataset.getMetadata();
-        const creationTime = Number(metadata.creationTime);
-        return creationTime && isResourceStale(creationTime);
+    const datasetTimePromises = datasets.map(async dataset => {
+      const [metadata] = await dataset.getMetadata();
```

Comment:
I think we may have slipped a bug in here-- I don't think "await" will actually work in an Array iterator like `map`/`forEach`/`filter`. The array iterator will return immediately and the promises will resolve later, but without a guarantee before we move on to the next code that depends on the values.

I tried rewriting this function with a `for of`, and I think this will do what we want:

```js
async function deleteDatasets() {
  const [datasets] = await bigquery.getDatasets({
    filter: `labels.${GCLOUD_TESTS_PREFIX}`,
  });

  for (const dataset of datasets) {
    const [metadata] = await dataset.getMetadata();
    const creationTime = Number(metadata.creationTime);

    if (isResourceStale(creationTime)) {
      try {
        await dataset.delete({force: true});
      } catch (e) {
        console.log(`dataset(${dataset.id}).delete() failed`);
        console.log(e);
      }
    }
  }
}
```

Note that I changed the call for "exists()" to just console.log the error if we get one.

Reply:
Okay, thanks!